src/rotation.zig

```diff
@@ -507,13 +548,40 @@ pub const RotationModes = struct {
 		pub const id: []const u8 = "stairs";
 		var modelIndex: u16 = 0;
 
-		fn subBlockMask(x: u1, y: u1, z: u1) u8 {
+		inline fn subBlockMask(x: u1, y: u1, z: u1) u8 {
 			return @as(u8, 1) << ((@as(u3, x)*2 + @as(u3, y))*2 + z);
 		}
-		fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
+		inline fn hasSubBlock(stairData: u8, x: u1, y: u1, z: u1) bool {
 			return stairData & subBlockMask(x, y, z) == 0;
 		}
 
+		fn rotateZ(data: u16) u16 {
+			comptime var rotateTable: [256]u8 = undefined;
+			@setEvalBranchQuota(10_000);
+			comptime for(0..256) |old| {
+				var new: u8 = 0;
+
+				for(0..2) |i| for(0..2) |j| for(0..2) |k| {
+					const sin: f32 = @sin(std.math.pi/2.0);
+					const cos: f32 = @cos(std.math.pi/2.0);
+
+					const x: f32 = (@as(f32, @floatFromInt(i)) - 0.5)*2.0;
+					const y: f32 = (@as(f32, @floatFromInt(j)) - 0.5)*2.0;
+					const z: f32 = (@as(f32, @floatFromInt(k)) - 0.5)*2.0;
```

Comment:
There is no need to include the calculations for `z` here, since it doesn't change.

Reply:
This covers all possible subblock combinations, to cover only changes in XY plane I would have to separately handle top and bottom half of the block. One is represented by odd bits, other by even bits. How would you implement it then?

Reply:
This covers all possible subblock combinations, to cover only changes in XY plane I would have to separately handle top and bottom half of the block. One is represented by odd bits, other by even bits. How would you implement it then?

Reply:
I would propose just this, and removing the then unused variables:
```diff
- subBlockMask(rX, rY, rZ);
+ subBlockMask(rX, rY, @intCast(k));
```

Reply:
Ok, I wonder if we maybe should restore angle parameter if we want to have table for 90 and 180, 270 rotations.

Reply:
Yeah, it might make sense to do. (probably best to do as an enum with `@"0"` `@"90"`, ... )